query/statement_rewriter.go

```diff
@@ -69,25 +69,39 @@ func rewriteShowFieldKeyCardinalityStatement(stmt *influxql.ShowFieldKeyCardinal
 
 	return &influxql.SelectStatement{
 		Fields: []*influxql.Field{
-			{
+			&influxql.Field{
 				Expr: &influxql.Call{
 					Name: "count",
 					Args: []influxql.Expr{
 						&influxql.Call{
 							Name: "distinct",
-							Args: []influxql.Expr{&influxql.VarRef{Val: "_fieldKey"}},
+							Args: []influxql.Expr{&influxql.VarRef{Val: "fieldKey"}},
 						},
 					},
 				},
 				Alias: "count",
 			},
 		},
-		Sources:    rewriteSources2(stmt.Sources, stmt.Database),
 		Condition:  stmt.Condition,
 		Dimensions: stmt.Dimensions,
-		Offset:     stmt.Offset,
-		Limit:      stmt.Limit,
 		OmitTime:   true,
+		Sources: influxql.Sources{
+			&influxql.SubQuery{
+				Statement: &influxql.SelectStatement{
+					Fields: influxql.Fields([]*influxql.Field{
+						{Expr: &influxql.VarRef{Val: "fieldKey"}},
+						{Expr: &influxql.VarRef{Val: "fieldType"}},
+					}),
+					Sources:    rewriteSources(stmt.Sources, "_fieldKeys", stmt.Database),
+					Condition:  rewriteSourcesCondition(stmt.Sources, nil),
+					Offset:     stmt.Offset,
```

Comment:
Feels like offset/limit should be either not accepted, or should go on the outer query?

Reply:
Outer query, I believe.  Good catch.